### PR TITLE
Use evented dataclass for colormap

### PR DESCRIPTION
**Pull request by sofroniewn**
# Description
This PR follows on from the other evented dataclass PRs using it now for the colormap.

## Type of change

- [x] Refactor (non-breaking change which fixes an issue)

### CODE REVIEW

**Comment by sofroniewn** [comment]
I've update this PR. It still includes the changes from #1914 which should be merged first, but the changes on the colormap are located to a single file, so it shouldn't be too hard to review.

File: napari/utils/colormaps/colormap.py [hunk 1]
```diff
@@ -21,41 +22,41 @@ class ColormapInterpolationMode(StringEnum):
     ZERO = auto()
 
 
+@evented_dataclass
 class Colormap:
     """Colormap that relates intensity values to colors.
 
-    Parameters
+    Attributes
```
Comment (tlambert03) [inline napari/utils/colormaps/colormap.py:29]: I get that the distinction is vague when they are both parameters (in the `__init__`) and attributes (of the object)... just curious if this change was discussed/decided somewhere previously?

File: napari/utils/colormaps/colormap.py [hunk 2]
```diff
@@ -21,41 +22,41 @@ class ColormapInterpolationMode(StringEnum):
...
-        self._interpolation = ColormapInterpolationMode(interpolation)
-        if controls is None:
+    def __post_init__(self):
+        if len(self.controls) == 0:
             n_controls = len(self.colors) + int(
                 self._interpolation == ColormapInterpolationMode.ZERO
             )
             self.controls = np.linspace(0, 1, n_controls)
-        else:
-            self.controls = np.asarray(controls)
```
Comment (tlambert03) [inline napari/utils/colormaps/colormap.py:55]: don't want to coerce `self.controls` with `np.asarray` anymore in case something else is passed?

**Review by tlambert03 — APPROVED** [review]
yep! looks good.  (only minor question about the fact that controls is no longer coerced to np.array during the init)

File: napari/utils/colormaps/colormap.py (hunk 1 above)
Comment (sofroniewn) [reply napari/utils/colormaps/colormap.py:29]: @jni mentioned it here https://github.com/napari/napari/pull/1911#discussion_r531258032, but the discussion was hardly long ;-) We may want to have both as they can diverge, as in #1917 for example.

File: napari/utils/colormaps/colormap.py (hunk 2 above)
Comment (sofroniewn) [reply napari/utils/colormaps/colormap.py:55]: yes, will do - good call!

Comment (sofroniewn) [reply napari/utils/colormaps/colormap.py:55]: done

File: napari/utils/colormaps/colormap.py (hunk 1 above)
Comment (sofroniewn) [reply napari/utils/colormaps/colormap.py:29]: I'll merge leaving as `Attributes` for now, but we can have a followup PR standardizing this for all the evented_data classes at some point in the near future

**Comment by sofroniewn** [comment]
I have made the requested changes. Will merge once tests pass

**Pull request merged**